addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -578,7 +578,7 @@ var FieldMany2One = AbstractField.extend({
      * @param {OdooEvent} ev
      */
     _onInputKeyup: function (ev) {
-        if (ev.which === $.ui.keyCode.ENTER) {
+        if (ev.which === $.ui.keyCode.ENTER || ev.which === $.ui.keyCode.TAB) {
```

Comment:
@aab-odoo 
Actually, replacing this line by `if (ev.which === $.ui.keyCode.TAB) {` does not break any test. 
So this behaviour is untested even for the ENTER key.

See TODO in the commit diff: with @adr-odoo we couldn't seem to replicate what seems to be the exact firing of events triggering the bug.